src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -1408,16 +1401,14 @@ private void PurgeDocById(string id)
         private void PurgeExpired(object state)
         {
             var cnt = 0L;
-            ThreadSafety.DoLockedBridge(err =>
+            LiteCoreBridge.Check(err =>
             {
                 CheckOpen();
                 cnt = Native.c4db_purgeExpiredDocs(_c4db, err);
                 Log.To.Database.I(Tag, "{0} purged {1} expired documents", this, cnt);
-                SchedulePurgeExpired(TimeSpan.FromSeconds(1));
-                if (err->code>0 && err->domain>0)
-                    return false;
-                return true;
+                return err;
```

Comment:
Why are you returning err?  You don't need to return anything.

Reply:
it complained if I don't return anything.

Reply:
Ok based on offline discussion it should be err.code == 0